### ISSUE TITLE
Warnings for redefining built-ins

### ISSUE THREAD

**Issue by treed** [koalaman/shellcheck#2446]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

test() {
    echo "Some stuff here that isn't actually testing anything"
}

NOPE=hi

if test -z $NOPE; then
    echo "No NOPE"
fi
```

which results in

```
Some stuff here that isn't actually testing anything
No NOPE
```

#### Here's what shellcheck currently says:

"No issues detected!"

#### Here's what I wanted or expected to see:

A warning about redefining a built-in would be a nice addition.

(And yes, I realize I was kinda dumb when naming a function test, but I guess that's what linters are for.)
